Documentation/linux-install-diagnosis.md

```diff
@@ -0,0 +1,144 @@
+distribution that some required packages have not yet been created
+or been moved out of Incoming.
+The following information may help to resolve the situation:
+
+The following packages have unmet dependencies:
+ dotnet-sdk-2.2 : Depends: aspnetcore-runtime-2.2 (>= 2.2.5) but it is not going to be installed
+                  Depends: dotnet-runtime-2.2 (>= 2.2.5) but it is not going to be installed
+E: Unable to correct problems, you have held broken packages.
+```
+
+This means `aspnetcore-runtime-2.2` and `dotnet-runtime-2.2` failed to install, but unfortunately it doesn't say why. You *can* follow the chain by running `apt-get install` for each unmet dependency, but it's usually safe to assume this error is because `dotnet-runtime-deps-2.2` failed to install.
+
+The following commands give more info about the error and the current state of your machine. Please run them and include the output in your issue report:
+
+```
+# This shows the package repositories you have configured. (Source: https://askubuntu.com/a/741948)
+grep -r --include '*.list' '^deb ' /etc/apt/sources.list /etc/apt/sources.list.d/
+
+# These commands show processor and Linux distribution information.
+uname -a
+cat /etc/os-release
+
+# This command shows why dotnet-runtime-deps-2.2 can't install.
+apt-get install dotnet-runtime-deps-2.2
+
+# You can copy this entire block of text and paste it into your terminal for convenience.
+```
+
+Next are common issues and resolutions:
+
+### Mismatched package repository vs. Linux distro/version
+The list of package repositories should include a `microsoft-prod.list` line, like this one:
+
+```
+/etc/apt/sources.list.d/microsoft-prod.list:deb [arch=amd64] https://packages.microsoft.com/ubuntu/18.10/prod cosmic main
+```
+
+Notice `ubuntu/18.10` in that example. That indicates that repository is meant for Ubuntu 18.04. If you try to use it on Ubuntu 19.04, for example, installation will fail.
+
+`cat /etc/os-release` shows you info about the distribution and version you're using:
+
+```
+NAME="Ubuntu"
+VERSION="19.04 (Disco Dingo)"
+...
+```
+
+To fix this, make sure the "Linux Distribution" dropdown has the correct selection and run the steps again.
+
+If the dropdown doesn't have an entry for your distribution and version, check if it is listed as supported at [release-notes/2.2/2.2-supported-os.md](../release-notes/2.2/2.2-supported-os.md). If it is supported, please file an issue to report the missing entry. Packages from a similar distribution or version may work.
+
+
+## `W: Failed to fetch https://packages.microsoft.com/ubuntu/14.04/prod/dists/trusty/main/binary-amd64/Packages Hash Sum mismatch`
+
+This indicates a service outage in the Microsoft repository, not limited to .NET Core. Please file an issue so we can have it fixed as soon as possible!
+
+---
+
+
+## Alternative install methods
+
+The Microsoft package repository is the preferred way to install .NET Core, but there are alternatives you can use as a workaround.
```

Comment:
I'm conflicted about the 'preferred' statement here. I can see us moving to a position where Snaps are the preferred method in a relatively short time frame. Perhaps say "is the most common way to install"?